### PR TITLE
updated the eslint and prettier and closes #26

### LINKED ISSUE #26: Add a linter and code formatter to the docs
## Problem

Currently, in the docs directory, we do not have a coding standard enforced, so it is difficult to maintain code consistency.

## Description

I want to add the two npm modules [eslint](https://eslint.org/) and [prettier](https://prettier.io/) to the project as well as a script to automate linting and code formatting.

## Alternatives

 - N/A

## Implementation

- [x] I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by Cronus1007**
## Fixes
Fixes #26 by @JackieBinya 

## Checklist
 
- [ ] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: docs/.eslintrc.js [hunk 1]
```diff
@@ -1,29 +1,104 @@
 module.exports = {
   root: true,
-  parser: 'vue-eslint-parser',
-  plugins: ['gridsome'],
-  env: {
-    node: true,
-  },
+  plugins: ['prettier', 'node', 'jest', 'jsdoc'],
```
Comment (hugosolar) [inline docs/.eslintrc.js:3]: Why the `gridsome` eslint [plugin](https://eslint.gridsome.org/introduction/) is removed here? 
seems to be very useful to check gridsome specifics

File: docs/.eslintrc.js [hunk 2]
```diff
@@ -1,29 +1,104 @@
...
   extends: [
-    'plugin:gridsome/recommended',
```
Comment (hugosolar) [inline docs/.eslintrc.js:9]: same here, do you have any specific reason for removing gridsome specifics extends?

File: docs/.eslintrc.js [hunk 3]
```diff
@@ -1,29 +1,104 @@
...
   ],
+  env: {
+    node: true,
+    es6: true,
+  },
+  parserOptions: {
+    ecmaVersion: 2018,
+  },
   rules: {
-    'no-console': process.env.NODE_ENV === 'production' ? 'error' : 'off',
```
Comment (hugosolar) [inline docs/.eslintrc.js:15]: do you have any specific reason to remove these debug lines?

**Review by hugosolar — COMMENTED** [review]
Thank you so much for your contribution! 
I have a couple of questions below.
Also, I think you can explain a little bit your solution in the PR description.

thanks again

File: docs/.eslintrc.js (hunk 3 above)
Comment (Cronus1007) [reply docs/.eslintrc.js:15]: Have a look at this. https://github.com/vuejs/vue-cli/issues/2266 I think we should include this rule when this issue gets resolved.

File: docs/.eslintrc.js (hunk 1 above)
Comment (Cronus1007) [reply docs/.eslintrc.js:3]: yarn 2 issues revolving round the plugin. So there is a great need to remove `eslint-plugin-gridsome` and these plugins are self sufficient to manage all the specifics.

File: docs/.eslintrc.js (hunk 2 above)
Comment (Cronus1007) [reply docs/.eslintrc.js:9]: Same above.

File: docs/.eslintrc.js (hunk 3 above)
Comment (hugosolar) [reply docs/.eslintrc.js:15]: ok, makes perfect sense to me

**Review by hugosolar — APPROVED** [review]
Thank you so much for your contribution ⭐

**Pull request merged**